Automattic/jetpack
# SSO: Implement users table improvements

## Proposed changes:
Currently, and it saddens me to say, the SSO situation is quite dire. If you add a user using wp-admin while SSO is on, that user simply doesn't work and there is no way for site admins to figure out why. It's especially painful when you import a site and a get a bunch of users who can't sign in, and it's not clear why.

In this PR, I work on improving the messaging in wp-admin. Make it clear when a user only exists locally on the site's DB, make it clear when a user is invited but not yet accepted, and when the user is functioning well. I also added the ability to invite users to wp.com from inside wp-admin.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
1. On an Atomic, JN, or whatever non-simple site, install Jetpack.
2. Make sure SSO is on, it should be on by default.
3. Go to the users table. 
4. It should have the status of the user's connection "Connected", "Invite", or "Pending invite". 
5. Clicking "Invite" should invite the user and change their status to "Pending invite". 
6. When inviting a user, you should get a notice saying "Hooray! User was invited successfully!".
7. While in this state, change the nonce in the URL to something incorrect, no notices should render.
8. Click back to get the correct nonce back.
9. Change the `jetpack-sso-invite-user` query param value to something that is not `success` or `failed`. No notice should render.
10. Inspect a non-invited user's "Invite" button and change the `user_id` hidden field to something incorrect, invite the user, you should get a nice error.
